src/main/java/lotto/domain/Ranks.java

```diff
@@ -0,0 +1,42 @@
+package lotto.domain;
+
+import java.util.EnumMap;
+import java.util.List;
+
+public class Ranks {
+    private final EnumMap<Rank, Integer> ranks = new EnumMap<>(Rank.class);
+
+    private final int purchaseAmount;
+
+    public Ranks(List<Rank> ranks, int purchaseAmount) {
+        init();
+        this.purchaseAmount = purchaseAmount;
+        for (Rank rank : ranks) {
+            this.ranks.put(rank, countRank(rank));
+        }
+    }
+
+    private void init(){
+        for(Rank rank : Rank.values()) {
+            this.ranks.put(rank, 0);
+        }
+    }
+
+    private Integer countRank(Rank rank) {
+        return this.ranks.get(rank) + 1;
+    }
+
+    public EnumMap<Rank, Integer> getRanks() {
```

Comment:
구체화된 EnumMap보다 추상화된 Map타입으로 반환하면 어떤 장점이 있을까요?
```suggestion
    public Map<Rank, Integer> getRanks() {
```